opentelemetry-sdk/src/trace/mod.rs

```diff
@@ -112,8 +110,10 @@ mod tests {
         span.set_attribute(KeyValue::new("attribute1", "value1"));
         span.add_event("test-event".to_string(), vec![]);
         span.set_status(Status::error("cancelled"));
-        drop(span);
-        provider.force_flush();
+        span.end();
```

Comment:
Let's also test implicitly drop span will end it

Reply:
that is covered in others tests in this file itself. So we have coverage for explicit end() and implicit via drop() both.